source-hubspot-native/source_hubspot_native/api.py

```diff
@@ -51,16 +83,27 @@ async def fetch_page(
     cutoff: datetime,
 ) -> AsyncGenerator[CRMObject | str, None]:
 
-    url = f"{HUB}/crm/v3/objects/{cls.NAME}"
-    properties = await fetch_properties(log, cls, http)
-    property_names = ",".join(p.name for p in properties.results if not p.calculated)
+    url = f"{HUB}/crm/v3/objects/{cls.PROPERTY_SEARCH_NAME}"
 
-    input = {
+    if cls.IGNORE_PROPERTY_SEARCH is True:
+        input = {
         "associations": ",".join(cls.ASSOCIATED_ENTITIES),
         "limit": 2,  # 50, # Maximum when requesting history. TODO(johnny).
-        "properties": property_names,
-        "propertiesWithHistory": property_names,
-    }
+        }
+        if len(cls.ASSOCIATED_ENTITIES) == 0:
+            del input['associations']
+    else: 
+        properties = await fetch_properties(log, cls, http)
+        property_names = ",".join(p.name for p in properties.results if not p.calculated)
+
+        input = {
+            "associations": ",".join(cls.ASSOCIATED_ENTITIES),
+            "limit": 2,  # 50, # Maximum when requesting history. TODO(johnny).
```

Comment:
A limit of 2 seems really small? Not sure if it was set at that before though.

Reply:
Yes, raised in 0b549e9f4cdfafdc3aadf8255959108dfb071aab to be the default value. Will leave johnny comments for future reference in case the limit number needs to be revised